File: boilerplate/java/order-service/src/main/java/com/example/orderservice/application/usecases/GetCurrentUserUseCaseImpl.java
package com.example.orderservice.application.usecases;

import com.example.orderservice.application.dtos.UserProfileResult;
import com.example.orderservice.domain.models.AuthenticationException;
import com.example.orderservice.domain.models.UserId;
import com.example.orderservice.domain.ports.UserRepository;

public class GetCurrentUserUseCaseImpl implements GetCurrentUserUseCase {

    private final UserRepository userRepository;

    public GetCurrentUserUseCaseImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public UserProfileResult execute(UserId userId) {
        var user = userRepository.findById(userId)
            .orElseThrow(() -> new AuthenticationException("AUTH_USER_NOT_FOUND", "User not found"));

        return new UserProfileResult(
            user.getId().getValue().toString(),
            user.getEmail().getValue(),
            user.getRoles(),
            user.isEnabled(),
            user.getCreatedAt(),
            user.getLastLoginAt()
        );
    }
}

File: boilerplate/nestjs/order-service/src/application/usecases/get-current-user.use-case.impl.ts
import { Injectable, Inject } from '@nestjs/common';
import { IGetCurrentUserUseCase } from './get-current-user.use-case.interface';
import { UserProfileResult } from '../dtos/user-order.dto';
import { UserId } from '../../domain/models/user-id.value-object';
import { IUserRepository } from '../../domain/ports/user-repository.port';
import { AuthenticationException } from '../../domain/exceptions/auth.exception';

@Injectable()
export class GetCurrentUserUseCaseImpl implements IGetCurrentUserUseCase {
    constructor(
        @Inject('IUserRepository')
        private readonly userRepository: IUserRepository,
    ) {}

    async execute(userId: UserId): Promise<UserProfileResult> {
        const user = await this.userRepository.findById(userId);

        if (!user) {
            throw new AuthenticationException('AUTH_USER_NOT_FOUND', 'User not found');
        }

        return {
            userId: user.id.getValue(),
            email: user.email.getValue(),
            roles: user.roles,
            enabled: user.enabled,
            createdAt: user.createdAt,
            lastLoginAt: user.lastLoginAt,
        };
    }
}
